### PR TITLE
Merge user-request-location into Master

### PR DESCRIPTION
**Pull request by sehoven**
Added functionality for the user to select the start and end locations on a map for a request.

### CODE REVIEW

File: app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java [hunk 1]
```diff
@@ -6,56 +6,154 @@
...
+/*
+ The code for incrementing/decrementing the fare while holding down
+ the up and down arrows is based on: https://goo.gl/zKpYnX
+ Author: Yar
+ Retrieved on: November 5th, 2016
+  */
 
 public class MakeRequestActivity extends AppCompatActivity {
 
+    private static final int DEFAULT_REQ_CODE = 1;
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java]: Can you comment on what default req code is here?

File: app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java [hunk 2]
```diff
@@ -6,56 +6,154 @@
...
+
     private Location start = null;
     private Location end = null;
     private int fareEstimated = -1;
 
+    private Handler repeatUpdateHandler = new Handler();
+    private boolean autoIncrement = false;
+    private boolean autoDecrement = false;
+
+    class RepeatUpdater implements Runnable {
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java:49]: Java doc for class? comments?

File: app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java [hunk 3]
```diff
@@ -6,56 +6,154 @@
...
+        }
+    }
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_make_request);
         setTitle("New Request");
-
+        setButtons();
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java]: What does this line do? (I may just have no idea what is going on?)

File: app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java [hunk 4]
```diff
@@ -6,56 +6,154 @@
...
         activity.getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_PAN);
     }
 
+    // TODO instead of location tuples, allow user to view start-to-end path on a map
+    // from: http://stackoverflow.com/questions/14292398/how-to-pass-data-from-2nd-activity-to-1st-activity-when-pressed-back-android
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java]: Add this to code re use wiki

File: app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java [hunk 5]
```diff
@@ -6,56 +6,154 @@
...
+    // author: ρяσѕρєя K
+    // retrieved on: November 7th, 2016
+    public void onActivityResult(int requestCode, int resultCode, Intent intent) {
+        super.onActivityResult(requestCode, resultCode, intent);
+        if(requestCode == DEFAULT_REQ_CODE) {
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java]: What? Why is this a thing? I made a comment about your declaration above... maybe I will find more below...

File: app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java [hunk 6]
```diff
@@ -6,56 +6,154 @@
...
+                tv = (TextView) findViewById(R.id.textView_end);
+                tv.setText("End: (" + String.valueOf(end.getLatitude()) + ", " + String.valueOf(end.getLongitude()) + ")");
+            }
+        }
+    }
+
+    /**
+     * Set the buttons for incrementing and decrementing the fare estimate by the user holding it down
+     */
+    private void setButtons() {
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java:94]: This looks good

File: app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java [hunk 7]
```diff
@@ -6,56 +6,154 @@
...
+            public boolean onLongClick(View view) {
+                autoDecrement = true;
+                repeatUpdateHandler.post(new RepeatUpdater());
+                return false;
+            }
+        });
+        fareDownButton.setOnTouchListener(new View.OnTouchListener() {
+            @Override
+            public boolean onTouch(View view, MotionEvent motionEvent) {
+                if((motionEvent.getAction() == MotionEvent.ACTION_UP
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java:131]: Can this if statement have a couple of comments? not really sure what is happening?

File: app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java [hunk 8]
```diff
@@ -6,56 +6,154 @@
...
+        }
+        if(end == null) {
+            end = new Location("");
+        } else {
+            bundle.putString("endLocation", new Gson().toJson(end));
+        }
+
+        Intent intent = new Intent(MakeRequestActivity.this, SetLocationsActivity.class);
+        intent.putExtras(bundle);
+        startActivityForResult(intent, DEFAULT_REQ_CODE);
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java]: still confused what default req code is

File: app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java [hunk 9]
```diff
@@ -65,20 +163,19 @@ public void chooseLocations(View view) {
      */
     public void onBackPressed() {
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java:171]: more java docs?

File: app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java [hunk 10]
```diff
@@ -104,6 +201,28 @@ public void estimateFare(View view) {
...
+        if(fareEstimated > 0) {
+            fareEstimated--;
+            TextView fareTextView = (TextView) findViewById(R.id.textView_fareEstimate);
+            fareTextView.setText(formatFare(fareEstimated));
+        }
+    }
+
+    /**
      * Format the fare we get as a double as a string to be printed on the screen.
      * This string will be preceded by a dollar sign (are we concerned about locale?)
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java:234]: can you please delete the locale thing

File: app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java [hunk 11]
```diff
@@ -0,0 +1,332 @@
...
+            }
+            if(intent.hasExtra("endLocation")) {
+                lastBundle.putString("endLocation",bundle.getString("endLocation"));
+            }
+        }
+
+        checkPermissions();
+
+        // Create an instance of GoogleAPIClient.
+        if (googleApiClient == null) {
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java:78]: THIS NEED TO BE PUT IN CODE AND WIKI

Comment (MandyMeindersma) [reply app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java:78]: FROM THE TUTORIAL

File: app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java [hunk 12]
```diff
@@ -0,0 +1,332 @@
...
+            GeoPoint geoPoint = new GeoPoint(locationPoint.getLatitude(), locationPoint.getLongitude());
+            map = (MapView) findViewById(R.id.map);
+            setLocationMarker(map, geoPoint);
+            IMapController mapController = map.getController();
+            mapController.setZoom(18);
+            mapController.setCenter(geoPoint);
+        }
+    }
+
+    // the following two functions come from: https://developer.android.com/training/location/retrieve-current.html
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java]: PLEASE ADD TO WIKI

File: app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java [hunk 13]
```diff
@@ -0,0 +1,332 @@
...
+            }
+
+            @Override
+            public void onMarkerDragEnd(Marker marker) {
+                locationPoint.setLatitude(marker.getPosition().getLatitude());
+                locationPoint.setLongitude(marker.getPosition().getLongitude());
+            }
+
+            @Override
+            public void onMarkerDragStart(Marker marker) {
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java:142]: onMarkerDragStart does what? Why is it empty?

File: app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java [hunk 14]
```diff
@@ -0,0 +1,332 @@
...
+
+            }
+        });
+
+        map.getOverlays().add(startMarker);
+        map.invalidate();
+    }
+
+    // this function comes from: https://developer.android.com/training/permissions/requesting.html
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java]: ADD IT TO THE WIKI

File: app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java [hunk 15]
```diff
@@ -0,0 +1,332 @@
...
+                    adb.setPositiveButton("OK", null);
+                    adb.show();
+                }
+                break;
+            }
+            case MY_PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION: {
+                // If request is cancelled, the result arrays are empty.
+                if (grantResults.length > 0
+                        && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    // permission was granted, yay! Do the
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java]: THIS IS JUST COMMENTS? sHOULD THEIR BE A todo IN HERE

File: app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java [hunk 16]
```diff
@@ -0,0 +1,332 @@
...
+                    // task you need to do.
+                } else {
+                    // permission denied, boo! Disable the
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java]: this is just comments? Should there be a TODO?

File: app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java [hunk 17]
```diff
@@ -0,0 +1,332 @@
...
+            }
+            case MY_PERMISSIONS_REQUEST_ACCESS_COARSE_LOCATION: {
+                // If request is cancelled, the result arrays are empty.
+                if (grantResults.length > 0
+                        && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    // permission was granted, yay! Do the
+                    // task you need to do.
+                } else {
+                    // permission denied, boo! Disable the
+                    // functionality that depends on this permission.
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java]: SAME AS ABOVE

File: app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java [hunk 18]
```diff
@@ -0,0 +1,332 @@
...
+            ActivityCompat.requestPermissions(activity,
+                    new String[]{Manifest.permission.ACCESS_COARSE_LOCATION},
+                    MY_PERMISSIONS_REQUEST_ACCESS_COARSE_LOCATION);
+            return;
+        }
+        lastLocation = LocationServices.FusedLocationApi.getLastLocation(googleApiClient);
+        latitude = lastLocation.getLatitude();
+        longitude = lastLocation.getLongitude();
+
+        // This code is from: https://github.com/MKergall/osmbonuspack/wiki/Tutorial_0
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java]: PUT ON WIKI

File: app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java [hunk 19]
```diff
@@ -0,0 +1,332 @@
...
+            locationPoint = new Location("");
+            locationPoint.setLatitude(geoPoint.getLatitude());
+            locationPoint.setLongitude(geoPoint.getLongitude());
+            setLocationMarker(map, geoPoint);
+        }
+        return false;
+    }
+
+    // from: http://stackoverflow.com/questions/14292398/how-to-pass-data-from-2nd-activity-to-1st-activity-when-pressed-back-android
+    // author: ρяσѕρєя K
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java:199]: ADD TO WIKI?

**Review by MandyMeindersma — COMMENTED** [review]
You did great Sarah I am really proud of you. You are an inspiration to us all. Sorry half of this is in caps

File: app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java (hunk 4 above)
Comment (sehoven) [reply app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java]: Added!

File: app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java (hunk 5 above)
Comment (sehoven) [reply app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java]: I added a comment for more clarification

File: app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java (hunk 9 above)
Comment (sehoven) [reply app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java:171]: I don't understand. I have java docs here? Where do I need more?

File: app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java (hunk 10 above)
Comment (sehoven) [reply app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java:234]: It gives me a warning if I don't have this which is why I put it in. Talked to Bennett and Kieter as well and we agreed that we will take it out.

File: app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java (hunk 13 above)
Comment (sehoven) [reply app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java:142]: OnMarkerDragListener is an abstract class. This method must exist but we don't need it to do anything.

File: app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java (hunk 15 above)
Comment (sehoven) [reply app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java]: I changed a bunch of this actually to make the positions work but there are no todos required anymore.

**Comment by sehoven** [comment]
@MandyMeindersma I made the changes you requested. Please review again.

**Pull request merged**